keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -483,6 +485,42 @@ Here we count all disruptions in the counter towards `.spec.backoffLimit`. All
 Pod failures caused by exceeding the configured limits or with non-zero exit code
 are categorized as user errors and result in termination of the entire job.
 
+#### Story 4
+
+As a machine learning user, ML frameworks allow scheduling of multiple pods.  
+The Job controller does not typically wait for terminating pods to be marked as failed.  Tensorflow and other ML frameworks may have a requirement that they only want Pods to be started once the other pods are fully terminated.  The following yaml can fit these needs:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  template:
+    spec:
+      containers:
+      - name: main-job-container
+        image: job-image
+        command: ["./program"]
+        resources:
+          requests:
+            memory: "10Gi"
+            ephemeral-storage: "10Gi"
+          limits:
+            memory: "10Gi"
+            ephemeral-storage: "100Gi"
+  backoffLimit: 3
+  podFailurePolicy:
+    delayPodRecreationUntilTerminal: true
+    rules:
```

Comment:
IIUC, the rules aren't relevant for the use case, so please delete. The presence of the new field set to `true` is enough to ensure the issue does not occur.